### DISCUSSION TITLE
Easiest way to open a webpage and send messages from JavaScript to C#?

### DISCUSSION THREAD

**Discussion by ashotjanibekyan** [AvaloniaUI/Avalonia#14029] [Q&A]
Suppose I want to open a webpage and when the user clicks on a button, I want to know about it in C#. CefSharp has something called [JavascriptMessageReceived](https://cefsharp.github.io/api/79.1.x/html/E_CefSharp_IWebBrowser_JavascriptMessageReceived.htm) which lets you send messages from JavaScript to C# but I don't think we can use CefSharp with Avalonia. I was able to open a webpage, execute a javascript code, or get results from a sync javascript method (using CefNet.Avalonia, or CefGlue.Avalonia) but couldn't receive messages.

Is this currently possible? What if I only care about Windows, for example?

**Comment by thevortexcloud** [comment]
I don't know much about CefSharp, but if you control the web site there are possibly other (and I would argue better) ways to do this. EG you could set up some sort of web hook or just poll for a result from the server. If the value you need the result from is truly only available from Javascript then I am not sure.

**Comment by strongelf** [comment]
I want to know the way of Calling C# function from Javascript in Cefnet.avalonia.

**Comment by kyuranger** [comment]
Maybe you can use this:
[Real-time ASP.NET with SignalR](https://dotnet.microsoft.com/en-us/apps/aspnet/signalr)

It can communicate between client and webpage.

**Reply by strongelf** [reply]
Thanks for your reply. As I known that Asp.net is a kind of web framework technology,but I want a desktop application coding by C sharp. The client panel can be embedded by chrome framework.
